src/main/java/org/spongepowered/common/mixin/core/statistic/MixinAchievement.java

```diff
@@ -0,0 +1,102 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.common.mixin.core.statistic;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import org.spongepowered.api.statistic.Statistic;
+import org.spongepowered.api.statistic.achievement.Achievement;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.translation.Translation;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.common.interfaces.statistic.IMixinAchievement;
+import org.spongepowered.common.text.SpongeTexts;
+import org.spongepowered.common.text.translation.SpongeTranslation;
+
+import java.util.Collection;
+import java.util.Set;
+
+@Mixin(net.minecraft.stats.Achievement.class)
+public abstract class MixinAchievement extends MixinStatBase implements IMixinAchievement {
+
+    @Shadow private String achievementDescription;
+    @Shadow public net.minecraft.stats.Achievement parentAchievement;
+
+    private Set<Achievement> childAchievements = Sets.newLinkedHashSet();
+
+    @Override
+    public Translation getDescription() {
+        return new SpongeTranslation(this.achievementDescription);
+    }
+
+    @Override
+    public Optional<Achievement> getParent() {
+        return Optional.fromNullable((Achievement) this.parentAchievement);
+    }
+
+    @Override
+    public Collection<Achievement> getChildren() {
+        return this.childAchievements;
+    }
+
+    @Override
+    public void addChild(Achievement achievement) {
+        this.childAchievements = ImmutableSet.<Achievement>builder()
+                .addAll(this.childAchievements)
+                .add(achievement)
+                .build();
+    }
+
+    @Override
+    public Optional<Statistic> getSourceStatistic() {
+        // TODO How to implement?
```

Comment:
Its intent is to return the backing statistic, eg. an achievement for crafting an Iron Sword is baked by the statistic tracking how many iron swords you have crafted. Some achievements aren't backed by statics though (eg. travel to the end).